src/spider/storage/mysql/MySqlStorage.cpp

```
@@ -80,6 +80,27 @@ auto get_sql_string(sql::SQLString const& str) -> std::string {
 
 // NOLINTEND
 
+auto task_language_to_string(spider::core::TaskLanguage language) -> std::string {
+    switch (language) {
+        case spider::core::TaskLanguage::Python:
+            return "python";
+        case spider::core::TaskLanguage::Cpp:
+            return "cpp";
+        default:
+            return "";
+    }
+}
+
+auto string_to_task_language(std::string const& language) -> spider::core::TaskLanguage {
+    if (language == "python") {
+        return spider::core::TaskLanguage::Python;
+    }
+    if (language == "cpp") {
+        return spider::core::TaskLanguage::Cpp;
+    }
+    return spider::core::TaskLanguage::Cpp;
+}
```

* Is there a better way to reflect the error on unknown language types? I'm not familiar with the current Spider core error handling. Maybe throw an exception since I don't see it's easy to integrate `Result` into the current APIs.
* The current behavior is a bit confusing: the default behavior for string conversion is to return an empty string, whereas the string-to-type conversion returns Cpp by default.
  * If we can't report the error, how about using "unknown" and thus we know it's an unknown type set intentionally.

---

* There was talking about using the `Results` for a long time, but since it was not available at the time of initial development, it is not used at that time, and we don't have any time for the conversion.
* I am ok with an `unknown`. However, I am not 100% sure the convert to empty string by default is good. The function should immediately report an error instead.

How about we start to adopt `Result` by making these two function to return one to enforce the user of these functions to do explicit error check. We can devise a plan for better error handling for the whole C++ project later?

---

Unfortunately now right now, but I don't think it is hard to convert it.